### ISSUE TITLE
button capture with shapes, images on click event

### ISSUE THREAD

**Issue by passenger94** [Shoes3/shoes3#240]
i noticed that you can only capture the left mouse button when working with the click event on shapes, image looks the same, maybe other elements.
Looks easy to fix (tried, it works (linux), just commenting _if (button == 1)_ here
https://github.com/Shoes3/shoes3/blob/master/shoes/ruby.c#L1090

But there must be a reason for that, any clues ?

``` ruby
Shoes.app width: 700 do
    shp = oval 25, 25, 100, fill: red
    shp.click do
        b,l,t = mouse
        info "button : #{b}\nleft : #{l}\ntop : #{l}"
    end
    Shoes.show_log
end
```

no more luck when trying from "outside"   

``` ruby
Shoes.app width: 700 do
    shp = oval 25, 25, 100, fill: red
    shp.click do
        b,l,t = input
        info "button : #{b}\nleft : #{l}\ntop : #{l}"
    end
    def input; mouse end
    Shoes.show_log
end
```

**Comment by dredknight** [comment]
It seems I am not the only one with this issue. Just stumbled on it today.

**Comment by ccoupe** [comment]
I've came across something else in this area , not all widgets supply the documented `|btn ,t ,l| ` some just pass in self.

**Comment by ccoupe** [comment]
I should describe what I think is happening in canvas.c (shoes_canvas_send_click) it calls shoes_canvas_send_click2(self, button, x, y, &clicked) which calls the  widget specific `shoes_XXXX_send_click` function  and  and if that return is not nil, then it invokes the callback to user's click passing only self - no btn, no left, no top.  Click doe not behave as documented

If I understand the code correctly, click is more like a  mouse_down and release  would be better named as 'clicked'. I 'm going to try defining the 'clicked` method in ruby.c to be the same C code as release.

**Comment by ccoupe** [comment]
If anyone knows how to attach click and release procs to a textblock (fragment) please speak up. I suspect this involved with link which changed from Shoes 2 to Shoes 3 after _why left.

**Comment by passenger94** [comment]
do you speak about something like :
`para "some text", link( "some interesting text", click: proc{ |self| ... }, "some other text"`

**Comment by ccoupe** [comment]
Closed in Shoes 3.3.2
